server/controllers/users/addUsers.js

```diff
@@ -0,0 +1,37 @@
+const { hash } = require('bcrypt');
+const { sign } = require('../../utils/jwt');
+const { addUsersQuery, checkEmail } = require('../../database/queries');
+const { boomify } = require('../../utils');
+const { addUserSchema } = require('../../utils/validation');
+
+const addUsers = async (req, res, next) => {
+  try {
+    const { userName, email, password, confirmPassword } = req.body;
+    await addUserSchema.validate(
+      { userName, email, password, confirmPassword },
+
+      {
+        abortEarly: false,
+      }
+    );
+    const { rows: user } = await checkEmail({ email });
+    if (user[0]) {
+      next(boomify(401, 'Login Error', 'You are registered'));
```

Comment:
If the user exists this message will be sent and continue to the line after it so you need to wrap your function into `if-else` conditions or to return this line

```suggestion
      return next(boomify(401, 'Login Error', 'You are registered'));
```